File: EOCV-Sim/src/main/java/com/github/serivesmejia/eocvsim/input/source/NullSource.java
/*
 * Copyright (c) 2026 Sebastian Erives
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 *
 */

package com.github.serivesmejia.eocvsim.input.source;

import com.github.serivesmejia.eocvsim.input.InputSource;
import org.opencv.core.Mat;

import javax.swing.filechooser.FileFilter;

public class NullSource extends InputSource {
    @Override
    public boolean init() {
        return true;
    }

    @Override
    public void reset() {

    }

    @Override
    public void close() {

    }

    Mat emptyMat = new Mat();

    @Override
    public Mat update() {
        return emptyMat;
    }

    @Override
    public void onPause() {

    }

    @Override
    public void onResume() {

    }

    @Override
    protected InputSource internalCloneSource() {
        return new NullSource();
    }

    @Override
    public FileFilter getFileFilters() {
        return null;
    }

    @Override
    public long getCaptureTimeNanos() {
        return System.nanoTime();
    }
}

File: EOCV-Sim/src/main/java/com/github/serivesmejia/eocvsim/input/source/NullSource.kt
/*
 * Copyright (c) 2026 Sebastian Erives
 * Licensed under the MIT License.
 */

package com.github.serivesmejia.eocvsim.input.source

import com.github.serivesmejia.eocvsim.input.InputSource
import org.opencv.core.Mat
import javax.swing.filechooser.FileFilter

class NullSource : InputSource() {

    override fun init(): Boolean = true

    override fun reset() {}

    override fun close() {}

    @Transient private val emptyMat = Mat()

    override fun update(): Mat = emptyMat

    override fun onPause() {}

    override fun onResume() {}

    override fun internalCloneSource(): InputSource = NullSource()

    override val fileFilters: FileFilter? = null
    override val captureTimeNanos: Long get() = System.nanoTime()


}

